### PR TITLE
Print failure info from metadata if workflow fails (QA-1184).

### PR DESCRIPTION
**Pull request by cahrens**
Example of how a failure would look:

image

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Check that the **Product Owner** has signed off on any user-facing changes
- [ ] **Submitter**: Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Database checks:
  * If PR includes new or changed db queries, include the explain plans in the description
  * Make sure liquibase is updated if appropriate
  * If doing a migration, take a backup of the
  [dev](https://console.cloud.google.com/sql/instances/terraform-qfarbdq3lrexxck5htofjs5z6m/backups?project=broad-dsde-dev&organizationId=548622027621)
  and
  [alpha](https://console.cloud.google.com/sql/instances/terraform-r4caezzc35c4tb7pgdhwkmme4y/backups?project=broad-dsde-alpha&organizationId=548622027621)
  DBs in Google Cloud Console
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell your tech lead (TL) that the PR exists if they want to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by cahrens** [comment]
jenkins retest

**Comment by cahrens** [comment]
jenkins retest

**Comment by cahrens** [comment]
jenkins retest

**Comment by cahrens** [comment]
jenkins retest

**Comment by cahrens** [comment]
jenkins retest

**Comment by cahrens** [comment]
This is an example of what prints out when there is a failure:

````
Error Message
[{"causedBy":[{"causedBy":[],"message":"stderr for job `test_count_variants.count_variants:NA:1` contained one of the `memory-retry-error-keys: [OutOfMemory,Killed]` specified in the Cromwell config. Job might have run out of memory."}],"message":"Workflow failed"}] "[Fail]ed" was not equal to "[Succeed]ed"
Stacktrace
      org.scalatest.exceptions.TestFailedException: [{"causedBy":[{"causedBy":[],"message":"stderr for job `test_count_variants.count_variants:NA:1` contained one of the `memory-retry-error-keys: [OutOfMemory,Killed]` specified in the Cromwell config. Job might have run out of memory."}],"message":"Workflow failed"}] "[Fail]ed" was not equal to "[Succeed]ed"
      at org.scalatest.matchers.MatchersHelper$.indicateFailure(MatchersHelper.scala:344)

…

**Comment by cahrens** [comment]
jenkins retest

**Comment by cahrens** [comment]
jenkins retest

**Comment by cahrens** [comment]
jenkins retest

File: automation/src/test/scala/org/broadinstitute/dsde/test/api/RawlsApiSpec.scala [hunk 1]
```diff
@@ -899,8 +902,9 @@ class RawlsApiSpec extends TestKit(ActorSystem("MySpec")) with AnyFreeSpecLike w
               status should not be("Running")
               metadata
             }
-
-            parseWorkflowStatusFromMetadata(notRunningMetadata) should be("Succeeded")
+            withClue(Option(getWorkflowFieldFromMetadata(notRunningMetadata, "failures")).getOrElse("No failures in metadata")) {
+              parseWorkflowStatusFromMetadata(notRunningMetadata) should be("Succeeded")
```
Comment (cjllanwarne) [inline automation/src/test/scala/org/broadinstitute/dsde/test/api/RawlsApiSpec.scala:906]: Didn't you just rename this function?

Comment (cahrens) [reply automation/src/test/scala/org/broadinstitute/dsde/test/api/RawlsApiSpec.scala:906]: No, I added getWorkflowFieldFromMetadata

**Review by salonishah11 — APPROVED** [review]
LGTM 👍

**Review by cjllanwarne — APPROVED** [review]

**Pull request merged**